artemis-jdbc-store/src/main/java/org/apache/activemq/artemis/jdbc/store/drivers/AbstractJDBCDriver.java

```diff
@@ -191,9 +191,18 @@ private void createTableIfNotExists(String tableName, String... sqls) throws SQL
             } else {
                try (Statement statement = connection.createStatement();
                      ResultSet cntRs = statement.executeQuery(sqlProvider.getCountJournalRecordsSQL())) {
-                  if (rs.next() && rs.getInt(1) > 0) {
+                  int rows;
+                  if (cntRs.next() && (rows = cntRs.getInt(1)) > 0) {
                      logger.tracef("Table %s did exist but is not empty. Skipping initialization.", tableName);
+                     if (logger.isDebugEnabled()) {
+                        final long expectedRows = Stream.of(sqls).map(String::toUpperCase).filter(sql -> sql.contains("INSERT INTO")).count();
```

Comment:
Are you sure you meant this INSERT to only happen if debug?

Reply:
Nope, I meant that the check should be performed only with debug log level to help spotting weird behavious.
It is not the most robust check ever, but it is pretty helpfull for HA (for example).

Reply:
@clebertsuconic PLease do not merge it yet...because on DB2 it is creating some problems